components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -89,8 +91,19 @@ public class SCIMCommonUtils {
      */
     private static ThreadLocal<Boolean> threadLocalIsManagedThroughSCIMEP = new ThreadLocal<>();
 
+    /**
+     * ThreadLocal to indicate if the agent flow is managed through SCIM.
+     */
+    private static ThreadLocal<Boolean> threadLocalIsAgentFlowContextThroughSCIM = new ThreadLocal<>();
```

Comment:
I feel like the variable name is long and complex.
Few other suggestions:
- `threadLocalIsAgentSCIMRequest`
- `threadLocalIsSCIMAgentFlow`
- `threadLocalToTriggerAgentFlowInSCIM`